### PR TITLE
Fix tests on macOS

### PR DESCRIPTION
**Pull request by ttytm**
Best #182 first, then C types can be updated here as well before a merge.

Apparently math works different on mac, needed to add a little tolarance for some tests.

### Summary by CodeRabbit

- Refactor: Simplified the CI/CD pipeline by removing specific test configurations, enhancing the efficiency of the testing process.
- Test: Introduced platform-specific tolerance checks in various test functions, ensuring consistent behavior across different operating systems when comparing floating-point values.
- New Feature: Incorporated LAPACK and CBLAS functionality into the codebase for macOS, enhancing the software's capabilities on this platform.
- Refactor: Simplified code in several areas by using more concise syntax, improving code readability without altering functionality.
- Test: Enhanced test failure messages to include actual and expected values along with the tolerance, providing more detailed feedback during testing.
- Chore: Removed unused functions and updated function signatures to align with library usage, streamlining the codebase.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -69,14 +69,6 @@ jobs:
...
       - name: Execute Tests using Pure V Backend with Pure C Blas
         run: ~/.vmodules/vsl/bin/test ${{ matrix.flags }} --use-cblas
-        if: ${{ matrix.os != 'ubuntu-18.04' }}
-
-      - name: Execute Tests using Pure V Backend and Garbage Collection enabled
-        run: ~/.vmodules/vsl/bin/test ${{ matrix.flags }} --use-gc boehm
-
-      - name: Execute Tests using Pure V Backend with Pure C Blas and Garbage Collection enabled
-        run: ~/.vmodules/vsl/bin/test ${{ matrix.flags }} --use-cblas --use-gc boehm
-        if: ${{ matrix.os != 'ubuntu-18.04' }}
```
Comment (ttytm) [inline .github/workflows/ci.yml:79]: Removes apparently obsolete
- gc boehm is the default mode, but I might be missing something that it was called explicitly here. Else, there is no difference with the run above.
- ubuntu-18.04 is not in the matrix

File: vlas/lapack_macos.c.v [hunk 2]
```diff
@@ -0,0 +1,7 @@
+module vlas
+
+fn C.LAPACKE_dlange(norm byte, m int, n int, a &f64, lda int, work &f64) f64
+
+pub fn dlange(norm byte, m int, n int, a []f64, lda int, work []f64) f64 {
+	return unsafe { C.LAPACKE_dlange(norm, m, n, &a[0], lda, &work[0]) }
+}
```
[the remark this answers is not in the document]
Comment (ttytm) [reply vlas/lapack_macos.c.v:7]: The definition for this macos integration differs. There is no matrix layout as first argument.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -118,18 +110,7 @@ jobs:
...
-        run: ~/.vmodules/vsl/bin/test --use-cblas
-
-      - name: Execute Tests using Pure V Backend and Garbage Collection enabled
-        continue-on-error: true
-        run: ~/.vmodules/vsl/bin/test --use-gc boehm
-
-      - name: Execute Tests using Pure V Backend with Pure C Blas and Garbage Collection enabled
-        continue-on-error: true
-        run: ~/.vmodules/vsl/bin/test --use-cblas --use-gc boehm
+        run: ~/.vmodules/vsl/bin/test --use-cblas --skip-examples
```
Comment (ulises-jeremias) [inline .github/workflows/ci.yml:116]: why are we using the flag `--skip-examples`? 👀

Comment (ttytm) [reply .github/workflows/ci.yml:116]: Tests work, but not all examples build on mac currently.

Thought we can re-enable it when they do.

**Pull request merged**